src/main/java/org/checkerframework/specimin/JavaParserUtil.java

```diff
@@ -68,4 +73,64 @@ public static ResolvedReferenceType classOrInterfaceTypeToResolvedReferenceType(
       ClassOrInterfaceType type) {
     return type.resolve().asReferenceType();
   }
+
+    /**
+     * Searches the ancestors of the given node until it finds a class or interface node, and then
+     * returns the fully-qualified name of that class or interface.
+     *
+     * <p>This method will fail if it is called on a node that is not contained in a class or
+     * interface.
+     *
+     * @param node a node contained in a class or interface
+     * @return the fully-qualified name of the inner-most containing class or interface
+     */
+    @SuppressWarnings("signature") // result is a fully-qualified name or else this throws
+    public static @FullyQualifiedName String getEnclosingClassName(Node node) {
+      Node parent = getEnclosingClassLike(node);
```

Comment:
If we decide to handle `AnnotationDeclaration`, in `getEnclosingClassLike` (see my other comment), we should account for that case in this method as well.